spec/features/applications/show_spec.rb

```diff
@@ -0,0 +1,176 @@
+require "rails_helper"
+
+RSpec.describe "Application Show Page" do
+  before(:each) do
+    @application_1 = Application.create!(name: "John", street_address: "1234 ABC Lane", city: "Turing", state: "Backend", zipcode: "54321", description: "I love animals", status: "In Progress")
+    @application_with_no_pets = Application.create!(name: "John", street_address: "1234 ABC Lane", city: "Turing", state: "Backend", zipcode: "54321", description: "I love animals", status: "In Progress")
+
+    @shelter = Shelter.create!(foster_program: true, name: "Turing", city: "Backend", rank: 3)
+
+    @dog = @shelter.pets.create!(adoptable: true, age: 4, breed: "Golden Retriever", name: "Dog")
+    @cat = @shelter.pets.create!(adoptable: true, age: 1, breed: "Tabby", name: "Cat")
+    @hamster = @shelter.pets.create!(adoptable: true, age: 1, breed: "Tabby", name: "Hamster")
+
+    @application_pet_1 = ApplicationPet.create!(application_id: @application_1.id, pet_id: @dog.id)
+    @application_pet_2 = ApplicationPet.create!(application_id: @application_1.id, pet_id: @cat.id)
+  end
+
+  it "has application details" do
+    # User Story 1
+    visit "/applications/#{@application_1.id}"
+
+    expect(page).to have_content(@application_1.name)
+    expect(page).to have_content(@application_1.full_address)
+    expect(page).to have_content(@application_1.description)
+    expect(page).to have_link("#{@dog.name}")
+    expect(page).to have_link("#{@cat.name}")
+    expect(page).to have_content("In Progress")
+
+    click_link("#{@dog.name}")
+
+    expect(page.current_path).to eq("/pets/#{@dog.id}")
+
+    visit "/applications/#{@application_1.id}"
+    click_link("#{@cat.name}")
+
+    expect(page.current_path).to eq("/pets/#{@cat.id}")
+  end
+
+  describe "searching for pets for an application" do
+    it "has a search bar" do
+      # User Story 4
+      visit "/applications/#{@application_1.id}"
+
+      expect(page).to have_content("Add a Pet to this Application")
+      expect(page).to have_content("Search for Pets by name:")
+      expect(page).to have_button("Submit")
+      expect(page).to have_no_content("Hamster")
+      expect(page).to have_no_button("Adopt this pet")
+
+      fill_in("Search for Pets by name:", with: "Hamster")
+      click_button("Submit")
+
+      expect(page).to have_content("Add a Pet to this Application")
```

Comment:
Once you've tested for the presence of this text above, I don't think you need to verify it's still there, especially since you're verifying the current path of the applicationon line 59